core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -342,7 +342,7 @@ class LogSegment(val log: FileMessageSet,
    * The time this segment has waited to be rolled. If the first message in the segment does not have a timestamp,
    * the time is based on the create time of the segment. Otherwise the time is based on the timestamp of that message.
    */
-  def timeWaitedForRoll(now: Long) : Long= {
+  def timeWaitedForRoll(now: Long, messageTimestamp: Long) : Long= {
```

Comment:
Yes, I agree that we'd want the same timestamp to be logged there. We could store the output of `timeWaitedForRoll` and reuse it, but it doesn't seem worth it. Happy to leave it as is.